model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -32,8 +32,80 @@
 #include "models/input_data.h"
 #include "models/results.h"
 
+namespace {
+cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_prediction, float soft_threshold, int blur_strength) {
+    cv::Mat soft_prediction_blurred;
+    cv::blur(soft_prediction, soft_prediction_blurred, cv::Size{blur_strength, blur_strength});
+    assert(soft_prediction_blurred.channels() > 1);
+    cv::Mat hard_prediction{cv::Size{soft_prediction_blurred.rows, soft_prediction_blurred.cols}, CV_8UC1};
+    for (int i = 0; i < soft_prediction_blurred.rows; ++i) {
+        for (int j = 0; j < soft_prediction_blurred.cols; ++j) {
+            float max_prob = -std::numeric_limits<float>::infinity();
+            uint8_t max_id = 0;
+            for (int c = 0; c < soft_prediction_blurred.channels(); ++c) {
+                float prob = ((float*)soft_prediction_blurred.data)[
+                    i * soft_prediction_blurred.cols * soft_prediction_blurred.channels()
+                    + j * soft_prediction_blurred.channels()
+                    + c];
+                if (prob > max_prob) {
+                    max_prob = prob;
+                    max_id = c;
+                }
+            }
+            hard_prediction.at<uint8_t>(i, j) = max_id;
+        }
+    }
+    return hard_prediction;
+}
+}
+
 std::string SegmentationModel::ModelType = "Segmentation";
 
+SegmentationModel::SegmentationModel(std::shared_ptr<ov::Model>& model, const ov::AnyMap& configuration)
+        : ImageModel(model, configuration) {
+    auto blur_strength_iter = configuration.find("blur_strength");
+    if (blur_strength_iter == configuration.end()) {
+        if (model->has_rt_info("model_info", "blur_strength")) {
+            blur_strength = stoi(model->get_rt_info<std::string>("model_info", "blur_strength"));
+        }
+    } else {
+        blur_strength = blur_strength_iter->second.as<int>();
+    }
+    auto soft_threshold_iter = configuration.find("soft_threshold");
+    if (soft_threshold_iter == configuration.end()) {
+        if (model->has_rt_info("model_info", "soft_threshold")) {
+            soft_threshold = stof(model->get_rt_info<std::string>("model_info", "soft_threshold"));
+        }
+    } else {
+        soft_threshold = soft_threshold_iter->second.as<float>();
+    }
+    auto return_soft_prediction_iter = configuration.find("return_soft_prediction");
+    if (return_soft_prediction_iter == configuration.end()) {
+        if (model->has_rt_info("model_info", "return_soft_prediction")) {
+            return_soft_prediction = model->get_rt_info<std::string>("model_info", "return_soft_prediction") == "True";
```

Comment:
There is a conversion to bool tape as well as far as I remember. So, no need to compare strings.

Reply:
Ok, got it. OpenVINO is very separate as usual here in terms of representing the things )